pkg/endpoint/policy.go

```diff
@@ -53,7 +53,6 @@ func (e *Endpoint) HasBPFPolicyMap() bool {
 }
 
 // GetNamedPort returns the port for the given name.
-// Must be called with e.mutex NOT held
```

Comment:
Out of curiosity: why is this no longer the case? Is that an effect of this change or just a leftover from a previous change?

Reply:
This is a leftover from a previous change. The comment was necessary earlier when the implementation was taking a lock and/or the called functions depended on this one not holding the endpoint lock. This function does not take the endpoint lock, and the called functions explicitly state that the endpoint lock may be taken while them being called, e.g.:
```
// GetNamedPorts returns a copy of the named ports map. May return nil.
func (ipc *IPCache) GetNamedPorts() (npm types.NamedPortMultiMap) {
	// We must not acquire the IPCache mutex here, as that would establish a lock ordering of
	// Endpoint > IPCache (as endpoint.mutex can be held while calling GetNamedPorts)
	// Since InjectLabels requires IPCache > Endpoint, a deadlock can occur otherwise.
```